[3.6.x] converts query-basic to react

andrewkfiedler opened the pull request:
- Text, matchcase, time converted to react
 - Location, match types will come in a later commit

Right now, only text, matchcase, and the time component have been converted.  Will add on the others tomorrow.

andrewkfiedler committed:
Converts text, matchcase, and time to react in query-basic

 - Text, matchcase, time converted to react
 - Location, match types will come in a later commit

andrewkfiedler committed:
Remove unused query-time files

jordanwilking commented on ui-frontend/packages/catalog-ui-search/src/main/webapp/component/query-basic/query-basic.view.tsx:
```diff
@@ -260,59 +281,186 @@ function getFilterTree(model: any) {
+  }
+}) => {
+  const filters = []
+  if (basicFilter.anyText[0].value !== '') {
```
❓ Any chance that these array might be empty?

andrewkfiedler replied:
In this case no, but I need to go back and add some better typing to this file.

jordanwilking commented on ui-frontend/packages/catalog-ui-search/src/main/webapp/component/query-basic/query-basic.view.tsx:
```diff
@@ -260,59 +281,186 @@ function getFilterTree(model: any) {
+          <TextField
+            fullWidth
+            value={basicFilter.anyText ? basicFilter.anyText[0].value : ''}
+            placeholder={`Text to search for. Use "*" for wildcard.`}
```
❓ I'm may just be missing what this will look like, but is "Text to search for." intentional?

andrewkfiedler replied:
I copied the placeholder we had from the previous, open to changing it.

jordanwilking commented on ui-frontend/packages/catalog-ui-search/src/main/webapp/component/query-time/query-time.view.tsx:
```diff
@@ -0,0 +1,201 @@
+            </Grid>
+            <Grid item className="w-full pl-2">
+              <Autocomplete
+                fullWidth
```
📓 For some reason, Typescript complains about this value despite it existing in the docs and the definitions file.

andrewkfiedler replied:
Strange, I wasn't seeing an issue with it.

jordanwilking replied:
Yeah, I can't explain it unfortunately 🤷‍♂️. t was just showing up when I was adding the ts-ignores

jordanwilking commented on ui-frontend/packages/catalog-ui-search/src/main/webapp/component/query-time/query-time.view.tsx:
```diff
@@ -0,0 +1,201 @@
+                getOptionSelected={(option, value) =>
+                  option.value === value.value
+                }
+                onChange={(e, newValue) => {
```
✏️ Types

jordanwilking commented on ui-frontend/packages/catalog-ui-search/src/main/webapp/component/query-time/query-time.view.tsx:
```diff
@@ -0,0 +1,201 @@
+          <MenuItem value="RELATIVE">Relative</MenuItem>
+        </TextField>
+      </div>
+      {value ? (
```
Out of curiosity, is there a reason that you prefer a ternary to `{ value && (<Component />) }`?

andrewkfiedler replied:
My personal preference, it weirds me out to use the ANDing logic that way.  Definitely more terse than the ternary, but to me a bit less readable since it feels like it's taking advantage of a side effect.

jordanwilking reviewed (approved):
🎉

andrewkfiedler committed:
Rebase with ts error fixes

andrewkfiedler committed:
Adds location back to basic

andrewkfiedler committed:
Merge branch '3.6.x' of https://github.com/codice/ddf-ui into Basic

andrewkfiedler committed:
Fix ts issues, format

andrewkfiedler committed:
Add type matching back to basic

andrewkfiedler committed:
Clean up styling, make the time, location, type consistent

 - All checkbox based now, fix padding / spacing and add divider between settings

andrewkfiedler committed:
Remove now unnecessary code

andrewkfiedler committed:
Add execute search on enter in keyword

andrewkfiedler committed:
Add autofocus on text input to query-basic

andrewkfiedler committed:
Update filter builder to insert * for blank on anyText

andrewkfiedler committed:
Fix ts errors

andrewkfiedler commented:
![Screen Shot 2020-09-17 at 12 30 00 AM](https://user-images.githubusercontent.com/11984853/93434404-f5b2f280-f87c-11ea-852a-bb1af02dc123.png)
 
Was able to recreate an issue found in the hero during a production build, oddly enough doesn't show in dev builds.

I'm pretty sure this is an existing issue in the 3.6.x branch, not sure when it got in, but I'll endeavor to solve it.

jlcsmith commented on ui-frontend/packages/catalog-ui-search/src/main/webapp/component/filter-builder/filter-builder.tsx:
```diff
@@ -29,30 +29,52 @@ type Props = {
+      filter.filters[0].type === 'ILIKE' &&
+      filter.filters[0].value === ''
+    ) {
+      filter.filters[0].value = '*'
```
It looks like basic search is using "%" rather than "*".  Not sure why but probably should keep it consistent

andrewkfiedler replied:
That transformation will happen during the normalization, so what we send over the wire is still %.

andrewkfiedler committed:
Rollback typescript to version 3.8.3, lock down tslib to 2.0.1

 - 3.9 typescript introduces type error in addition to blatent ignores, but unfortunately it has some issues, see https://github.com/microsoft/TypeScript/issues/39977

jlcsmith commented:
🎉 hero successful 🎉
